python/pyspark/sql/functions.py

```diff
@@ -116,7 +114,7 @@ def __init__(self, func, returnType):
 
     def _create_judf(self):
         f = self.func  # put it in closure `func`
-        func = lambda _, it: imap(lambda x: f(*x), it)
+        func = lambda _, it: map(lambda x: f(*x), it)
```

Comment:
Will this cause a performance regression for 2.x users?  Can we replace the `imap` with a generator expression instead?